### PR TITLE
Add developer experience topic

### PR DESCRIPTION
**Pull request by jofaval**
### Please confirm this pull request meets the following requirements:

- [X] I followed the contributing guidelines: <https://github.com/github/explore/blob/main/CONTRIBUTING.md>.
- [X] I have no affiliation with the project I am suggesting (as a maintainer, creator, contractor, or employee).

### Which change are you proposing?

  - [ ] Suggesting edits to an existing topic or collection
  - [X] Curating a new topic or collection
  - [ ] Something that does not neatly fit into the binary options above

---

### Curating a new topic or collection

- [X] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [X] All required fields in my `index.md` conform to the Style Guide and API docs: <https://github.com/github/explore/tree/main/docs>

> My proposed changes are to add two topics, `developer-experience` and it's abbreviation `dx`, both with the same information since they refer to the same concept.
---

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: topics/developer-experience/index.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+---
+display_name: Developer experience
+related: dx
```
Comment (smashwilson) [inline topics/developer-experience/index.md:3]: You don't need to copy the entire file, we have a mechanism for this! Please convert this into an ["alias"](https://github.com/github/explore/blob/main/docs/API.md#aliases) and delete the other file.

**Review by smashwilson — CHANGES_REQUESTED** [review]

File: topics/developer-experience/index.md (hunk 1 above)
Comment (jofaval) [reply topics/developer-experience/index.md]: You're right! Thank you for clarifying it, it's done now.

**Review by smashwilson — APPROVED** [review]

**Comment by smashwilson** [comment]
Perfect! Thanks. 🙇🏻

**Comment by smashwilson** [comment]
Note that this won't be reflected _right_ away because I'm away from the computer where I can access the internal import tool right now. I'll set a reminder to take care of it once I'm back.

**Pull request merged**